src/components/Editable/useMultiline.ts

```diff
@@ -1,17 +1,24 @@
 import { useCallback, useLayoutEffect, useState } from 'react'
-import { shallowEqual, useSelector } from 'react-redux'
+import { useSelector } from 'react-redux'
 import SimplePath from '../../@types/SimplePath'
-import State from '../../@types/State'
-import useSelectorEffect from '../../hooks/useSelectorEffect'
-import editingValueStore from '../../stores/editingValue'
+import equalPath from '../../util/equalPath'
 
 /** Returns true if the element has more than one line of text. */
-const useMultiline = (contentRef: React.RefObject<HTMLElement>, simplePath: SimplePath, isEditing?: boolean) => {
+const useMultiline = (
+  contentRef: React.RefObject<HTMLElement>,
+  simplePath: SimplePath,
+  value: string,
+  isEditing?: boolean,
+) => {
   const [multiline, setMultiline] = useState(false)
   const fontSize = useSelector(state => state.fontSize)
   const showSplitView = useSelector(state => state.showSplitView)
   const splitPosition = useSelector(state => state.splitPosition)
 
+  // Check if the cursor is active.
+  // NOTE: We don't want to select the entire cursor path to skip invoking the layout effect too often.
+  const cursorActive = useSelector(state => equalPath(state.cursor, simplePath))
```

Comment:
`cursorActive` is not quite adequate to cover the behavior of the `useSelectorEffect`. We need to remeasure whenever the cursor changes, even if the cursor is not active on this thought. In the following example, if the cursor is moved from `aaaaaaaaaa` to `bbbbbbbbbb', then `cccccccccccc` may wrap and become multiline.

```
- aaaaaaaaaa
  - bbbbbbbbbb
    - cccccccccccc
```